File: Backend/BaseMgtService/Senior.AgileAI.BaseMgt.Application/DTOs/SignUpDTO.cs
namespace Senior.AgileAI.BaseMgt.Application.DTOs
{
    public class SignUpDTO
    {
        public required string FullName { get; set; }
        public required string Email { get; set; }
        public required string Password { get; set; }
        public required DateOnly BirthDate { get; set; }
        public required Guid Country_IdCountry { get; set; }
        
        

    }
}

File: agilemeets_mobile/lib/data/models/sign_up_dto.dart
import 'package:intl/intl.dart';

class SignUpDTO {
  final String? fullName;
  final String email;
  final String password;
  final DateTime? birthDate;
  final String? countryIdCountry;

  SignUpDTO({
    this.fullName,
    required this.email,
    required this.password,
    this.birthDate,
    this.countryIdCountry,
  });

  Map<String, dynamic> toJson() {
    return {
      'fullName': fullName,
      'email': email,
      'password': password,
      'birthDate': birthDate?.toIso8601String().split('T')[0],
      'country_IdCountry': countryIdCountry,
    };
  }
}
